### ISSUE TITLE
throwIfOrphans: true and input array elements order

### ISSUE THREAD

**Issue by lkallas** [philipstanislaus/performant-array-to-tree#18]
Consider a tree with 4 elements like this:

```javascript
{
  "id": "root",
  "parent_id": null,
  "bar": "bar",
  "children": [
    {
      "id": "2",
      "parent_id": "root",
      "foo": "bar",
      "children": []
    },
    {
      "id": "1",
      "parent_id": "root",
      "foo": "bar",
      "children": [
        {
          "id": "1-1",
          "parent_id": "1",
          "foo": "bar",
          "children": []
        }
      ]
    }
  ]
}
```

As you may see there is multi level nesting in this tree.

When I break this tree into individual elements and try to build it with option `throwIfOrphans: true` it will throw. It should not, because there are actually no orphans.  

This issue comes from the order of items in the array passed into the `arrayToTree` function.

To reproduce:

```javascript
const tree = arrayToTree(
  [
    { id: '2', parent_id: 'root', foo: 'bar' },
    { id: '1-1', parent_id: '1', foo: 'bar' },
    { id: '1', parent_id: 'root', foo: 'bar' },
    { id: 'root', parent_id: null, bar: 'bar' },
  ],
  {
    id: 'id',
    parentId: 'parent_id',
    childrenField: 'children',
    dataField: null,
    throwIfOrphans: true,
  },
);
```  

Results: 
```bash
Error: The items array contains orphans that point to the following parentIds: [1]. These parentIds do not exist in the items array. Hint: prevent orphans to result in an error by passing the following option: { throwIfOrphans: false }
```

Now when you change the order of elements in the array it will not throw:

```javascript
const tree = arrayToTree(
  [
    { id: '2', parent_id: 'root', foo: 'bar' },
    { id: '1', parent_id: 'root', foo: 'bar' },
    { id: '1-1', parent_id: '1', foo: 'bar' },
    { id: 'root', parent_id: null, bar: 'bar' },
  ],
  {
    id: 'id',
    parentId: 'parent_id',
    childrenField: 'children',
    dataField: null,
    throwIfOrphans: true,
  },
);
```

Meaning that the input array order matters. The error is misleading in this particular case.

**Comment by philipstanislaus** [comment]
Thanks for this outstanding bug report. I'll get back to you shortly!

**Comment by philipstanislaus** [comment]
Fixed, released in v1.7.1. Thanks for your help!
